src/urbit/mar/chess/update.hoon

```diff
@@ -40,6 +40,11 @@
         :~  ['chessUpdate' [%s 'draw-declined']]
             ['gameID' [%s (scot %da game-id.upd)]]
         ==
+      %threefold-checkbox
+        %-  pairs:enjs
+        :~  ['chessUpdate' [%s 'threefold-checkbox']]
+            ['gameID' [%s (scot %da game-id.upd)]]
+        ==  
```

Comment:
A checkbox is an entirely frontend concept - there should be absolutely no mention of it in the backend at all. In my design for a solution to the threefold repetition bug, the checkbox I mentioned is just a frontend interface for communicating a particular piece of data: does the user want to automatically claim threefold repetition draws, or not? As a "yes/no" piece of information, this is a single boolean (or loobean) that the user needs to communicate to the `%chess` app.

The problem that this is solving is that the player receiving a position that's been repeated three times can choose to claim or not claim a draw from that position. However, the player who is creating a position that's been repeated three times must make the draw claim as he makes the move. Therefore, this "yes/no" decision should be provided with an accompanying move. This provides the inspiration for how we should design the interface on the backend: this is information that we should receive with a corresponding move.

What kind of move should this information be a part of? We have two options available: `chess-move` and `chess-action`. I leave it as an exercise for you to decide which one would be a better design.

Reply:
I was thinking the yes/no state of the checkbox needed to be separate from a move so that the backend would automatically claim the draw for a player who receives a repeated position and has the box checked. To do that, my assumption was that `%chess` would need to keep track of the state of the checkbox.

Reply:
Your idea also makes sense. In that case, you would need to add a new type of poke that sets the user's preference for setting this option to true/false for the game. It works well with my design, in that the default value of this setting on the backend would be part of the not-yet-implemented user preferences (e.g. auto-claim draws, practice board state, board visual settings, etc.).

However, the name should still definitely not include "checkbox" anywhere in it, as that's a frontend interface abstraction.